File: code/cpp/1282.cpp
# https://leetcode.com/problems/group-the-people-given-the-group-size-they-belong-to/

class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& groupSizes) {
        unordered_map<int, vector<int>> mp;
        vector<vector<int>> res;
        for (size_t i = 0; i < groupSizes.size(); i++) {
            if (groupSizes[i] == 1) {
                res.emplace_back();
                res.back().emplace_back(i);
            }
            else
                mp[groupSizes[i]].emplace_back(i);
        }
        for (auto e : mp) {
            for (size_t i = 0; i < e.second.size(); i++){
                if (i % e.first == 0)
                    res.emplace_back();
                res.back().emplace_back(e.second[i]);
            }
        }
        return res;
    }
};

File: code/python/1282.py
# https://leetcode.com/problems/group-the-people-given-the-group-size-they-belong-to/

class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        mp, res = {}, []
        for i, group in enumerate(groupSizes):
            mp[group] = mp.get(group, []) + [i]
        for key, val in mp.items():
            tmp = [val[i:i+key] for i in range(0, len(val), key)]
            res.extend(tmp)
        return res
